airflow/providers/amazon/aws/hooks/eks.py

```diff
@@ -0,0 +1,452 @@
+
+    def list_nodegroups(
+        self,
+        clusterName: str,
+        verbose: Optional[bool] = False,
+    ) -> List:
+        """
+        Lists all Amazon EKS Nodegroups associated with the specified cluster.
+
+        :param clusterName: The name of the Amazon EKS Cluster containing nodegroups to list.
+        :type clusterName: str
+        :param verbose: Provides additional logging if set to True.  Defaults to False.
+        :type verbose: bool
+
+        :return: A List of nodegroup names within the given cluster.
+        :rtype: List
+        """
+        eks_client = self.conn
+        api_call = partial(eks_client.list_nodegroups, clusterName=clusterName)
+
+        return self._list_all(api_call=api_call, response_key="nodegroups", verbose=verbose)
+
+    def _list_all(self, api_call: Callable, response_key: str, verbose: bool) -> List:
+        """
+        Repeatedly calls a provided boto3 API Callable and collates the responses into a List.
+
+        :param api_call: The api command to execute.
+        :type api_call: Callable
+        :param response_key: Which dict key to collect into the final list.
+        :type response_key: str
+        :param verbose: Provides additional logging if set to True.  Defaults to False.
+        :type verbose: bool
+
+        :return: A List of the combined results of the provided API call.
+        :rtype: List
+        """
+        name_collection = []
+        token = DEFAULT_PAGINATION_TOKEN
+
+        try:
+            while token != "null":
+                response = api_call(nextToken=token)
+                # If response list is not empty, append it to the running list.
+                name_collection += filter(None, response.get(response_key))
+                token = response.get("nextToken")
+
+            self.log.info("Retrieved list of %s %s.", len(name_collection), response_key)
+            if verbose:
+                self.log.info("%s found: %s", response_key.title(), name_collection)
+
+        except ClientError as e:
+            self.log.error(e.response["Error"]["Message"])
+            raise e
+        return name_collection
+
+    @contextmanager
+    def generate_config_file(
+        self,
+        eks_cluster_name: str,
+        pod_namespace: str,
+        pod_username: Optional[str] = DEFAULT_POD_USERNAME,
+        pod_context: Optional[str] = DEFAULT_CONTEXT_NAME,
+    ) -> str:
+        """
+        Writes the kubeconfig file given an EKS Cluster.
+
+        :param eks_cluster_name: The name of the cluster to create the EKS Managed Nodegroup in.
+        :type eks_cluster_name: str
+        :param pod_namespace: The namespace to run within kubernetes.
+        :type pod_namespace: str
+        :param pod_username: The username under which to execute the pod.
+        :type pod_username: str
+        :param pod_context: The name of the context access parameters to use.
+        :type pod_context: str
```

Comment:
Can you docs about return values? It is not obvious that this is a name for a temporary file.

Reply:
Nothing here has to do with a filename; it is a randomly generated name.  Those values are passed into the kubeconfig file.